### PR TITLE
Connect poll to backend

### PR DESCRIPTION
**Pull request by heeseop-an**
Jira: https://pro-planner.atlassian.net/browse/PLAN-19

- Connected `VotePage` to backend

To test, connect to mongoDB and change `currUser` in `Poll` to simulate the voting page with multiple users.

### CODE REVIEW

File: pro-planner-client/src/components/AddOptionForm.jsx [hunk 1]
```diff
@@ -1,28 +1,21 @@
...
-        option: newOption,
-        voteCount: 0,
-    }
+    const pollDocumentId = useSelector((state) => state.poll.pollsId)
 
     const handleAddOption = (e) => {
         e.preventDefault();
         const formattedNewOption = newOption.trim();
-        const options = Object.values(poll.options)
+        const options = poll.options || {};
```
Comment (rikili) [inline pro-planner-client/src/components/AddOptionForm.jsx:18]: Never seen this before, it's very elegant way to handle this!

File: pro-planner-client/src/redux/pollSlice.js [hunk 2]
```diff
@@ -39,6 +88,54 @@ const pollSlice = createSlice({
...
+        builder.addCase(getPollAsync.rejected, (state) => {
+            state.pollStatus = LOAD_STATUS.FAILED;
+        });
+
+        builder.addCase(addPollAsync.pending, (state) => {
+            state.pollStatus = LOAD_STATUS.LOADING;
+        });
+        builder.addCase(addPollAsync.fulfilled, (state, action) => {
+            state.polls = action.payload.polls;
+            console.log(state.polls)
```
Comment (rikili) [inline pro-planner-client/src/redux/pollSlice.js:110]: may want to remove this console.log

File: pro-planner-client/src/routes/VotePage.jsx [hunk 3]
```diff
@@ -1,9 +1,26 @@
...
+import React, {useEffect} from "react";
 import AddPollForm from "../components/AddPollForm";
 import Polls from "../components/Polls";
+import {getPollAsync} from "../redux/pollSlice";
+import {useDispatch, useSelector} from "react-redux";
+import {LOAD_STATUS} from "../constants";
 
 const VotePage = () => {
+
+    // const tripId = useLocation().pathname.split('/')[1]; TODO...
```
Comment (rikili) [inline pro-planner-client/src/routes/VotePage.jsx:10]: Should this be included for a merge to main?

**Review by rikili — COMMENTED** [review]
Seems mostly complete, but some debug stuff and testing code seem to need a bit of sorting out.

**Review by rikili — APPROVED** [review]
Looks good!

**Review by vincewu124 — APPROVED** [review]
looks good

**Pull request closed without merging**